File: C++Code/Hashing/subarray.cpp
#include <iostream>
#include<unordered_set>
using namespace std;
int sub_array(int arr[], int n, int sum)
{
    unordered_set<int> s;
    int preSum=0;
    for(int i=0;i<n;i++){
        preSum=preSum+arr[i];
        if(s.find(preSum)!=s.end()){
            return true;
        }
        if(preSum==0){
            return true;
        }
        s.insert(arr[i]);
    }
    return false;

}
int main()
{
    int arr1[] = {2, 3, 4, -7, 5, 4};
    int n1 = sizeof(arr1) / sizeof(arr1[0]);
    int sum = 0;
    cout<<sub_array(arr1, n1, sum)<<endl;
}

File: JavaScritpt/Arrays/Subarray.js
function Subarray(arr,sum,n){
    var i,Total=0;
    for(i=0;i<n;i++){
        Total=arr[i];
        for(let j=i+1;j<=n;j++){
            if(Total==sum){
                return console.log(i+""+(j));
            }      
            if(Total>sum || j==n){
                    break;            }
            
            Total=Total+arr[j];
                
    
    }
}
return console.log('Not Found')
}
let arr= [15, 2, 4, 8, 9, 5, 10, 23];
let n = arr.length;
let sum = 23;
Subarray(arr, n, sum);